### DISCUSSION TITLE
什么情况下 `viewMeta.data` 的类型才会是SimpleData？

### DISCUSSION THREAD

**Discussion by liqiuqiui** [antvis/S2#2923] [Q&A] [answered]
文档中仅给出`viewMeta.data`类型，并未做任何说明

**Comment by lijinke666** [comment, marked as answer]
这里应该是文档错误, 不会有 SimpleData 的情况

**Reply by liqiuqiui** [reply]
不仅是文档，实际上ViewMeta.data类型定义也是有SimpleData的
![image](https://github.com/user-attachments/assets/bc85e339-ed39-4b47-bfbc-d0f9c4a60d73)

**Reply by lijinke666** [reply]
是的, type 也有问题, 另外你在意的点是?

**Reply by liqiuqiui** [reply]
> 是的, type 也有问题, 另外你在意的点是?

因为实际使用中我需要拿到这个data 去做一些别的处理，怕没有考虑到simpledata的情况下会出错。所以才提出这个问题。

**Reply by lijinke666** [reply]
好的, 近期优化下 type 和文档, 你也可以增加下容错处理

**Comment by lijinke666** [comment]
https://github.com/antvis/S2/pull/2935
